src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -264,11 +308,34 @@ private SpoolMessage getSpoolMessageFromRs(long messageId, ResultSet rs) throws
         }
     }
 
+    /**
+     * This method checks for connection exists or not if not creates connection.
+     * @throws SQLException When fails to get Database Connection
+     */
+    synchronized void ensureConnection() throws SQLException {
+        if (dbConnection == null || dbConnection.isClosed()) {
+            try (LockScope ignored = LockScope.lock(dbConnectionLock.writeLock())) {
+                init();
+            }
+        }
+    }
+
+    boolean isConnectionHealthy(Connection dbConnection) {
+        try (Statement stmt = dbConnection.createStatement()) {
```

Comment:
can we use a prepared statement so we don't have to create the same SELECT 1 query every time?